src/components/signup/SignupState.js

```diff
@@ -224,6 +229,14 @@ const Signup = ({ navigation }: { navigation: any, screenProps: any }) => {
     checkTorusLogin()
     verifyStartRoute()
 
+    // Recognize registration method (page refresh case included)
+    const initialRegMethod = await AsyncStorage.getItem(GD_INITIAL_REG_METHOD)
+    const _regMethod = initialRegMethod
+      ? initialRegMethod
+      : get(navigation, 'state.routes[1].params.regMethod', 'selfCustody')
```

Comment:
could be prettier with destructuring

const [,activatedRoute] = state.routes
const { regMethod } = activatedRoute.params

Reply:
I think it is not critical